test/Broadway/EventSourcing/AggregateFactory/ReflectionAggregateFactoryTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Broadway\EventSourcing\AggregateFactory;
+
+use Broadway\Domain\DomainEventStream;
+use Broadway\EventSourcing\EventSourcedAggregateRoot;
+use Broadway\TestCase;
+
+/**
+ *
+ */
+final class ReflectionAggregateFactoryTest extends TestCase
+{
+    /**
+     * @var ReflectionAggregateFactory
+     */
+    private $factory;
+
+    public function setUp()
+    {
+        $this->factory = new ReflectionAggregateFactory();
+    }
+
+    /**
+     * @test
+     */
+    public function it_creates_instance_of_aggregate_with_private_constructor()
+    {
+        $aggregate = $this->factory->create(
+            MyTestAggregateRootForReflectionFactory::class,
+            new DomainEventStream([])
+        );
+
+        $this->assertInstanceOf(MyTestAggregateRootForReflectionFactory::class, $aggregate);
+    }
+
+    /**
+     * @test
+     * @expectedException \LogicException
+     * @expectedExceptionMessage Impossible to initialize "stdClass"
+     */
+    public function it_does_not_handle_weird_classes()
+    {
+        $this->factory->create(\stdClass::class, new DomainEventStream([]));
+    }
+}
+
+// TODO: separate folder with test aggregates?
```

Comment:
I think it makes the test more clear to keep them in the same file for now.
If we would get more complex tests, we might want to move them somewhere else.

For now I would say, remove this TODO and I'll merge this :)